addon/components/lpdc-plugin/lpdc-insert.ts

```diff
@@ -0,0 +1,94 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { AddIcon } from '@appuniversum/ember-appuniversum/components/icons/add';
+
+import { SayController } from '@lblod/ember-rdfa-editor';
+import { sayDataFactory } from '@lblod/ember-rdfa-editor/core/say-data-factory';
+import { addProperty } from '@lblod/ember-rdfa-editor/commands';
+import {
+  LPDC,
+  type LpdcPluginConfig,
+} from '@lblod/ember-rdfa-editor-lblod-plugins/plugins/lpdc-plugin';
+import { v4 as uuidv4 } from 'uuid';
+import { getCurrentBesluitRange } from '../../plugins/besluit-topic-plugin/utils/helpers';
+import { SRO } from '../../utils/constants';
+
+interface Args {
+  controller: SayController;
+  config: LpdcPluginConfig;
+}
+
+export default class LpdcPluginsInsertComponent extends Component<Args> {
+  AddIcon = AddIcon;
+
+  @tracked showModal = false;
+
+  get controller() {
+    return this.args.controller;
+  }
+
+  @action
+  openModal() {
+    this.controller.focus();
+    this.showModal = true;
+  }
+
+  @action
+  closeModal() {
+    this.showModal = false;
+  }
+
+  get isButtonDisabled() {
+    return !getCurrentBesluitRange(this.controller);
+  }
+
+  @action
+  onLpdcInsert(lpdc: LPDC) {
+    const rdfaId = uuidv4();
+
+    const uri = lpdc.uri;
+    const name = lpdc.name;
+
+    const currentBesluitRange = getCurrentBesluitRange(this.controller);
+
+    const resource =
+      (currentBesluitRange &&
+        'node' in currentBesluitRange &&
+        (currentBesluitRange.node.attrs.subject as string)) ||
+      undefined;
+
+    if (!resource) {
+      throw new Error('No besluit found in selection');
+    }
+
+    this.controller.withTransaction(
+      (tr) => {
+        const node = this.controller.schema.node(
+          'inline_rdfa',
+          {
+            rdfaNodeType: 'literal',
```

Comment:
The way I read [the spec](https://github.com/Informatievlaanderen/OSLOthema-slimmeRaadpleegOmgeving#lpdc-codes) is that this should be a resource. I'm guessing since it isn't clear where the `dcterms:source` in their example comes from, this is only necessary if we want to add extra triples to the otherwise externally defined resource? But literal seems wrong according to their example.